### ISSUE TITLE
Selection is not lost when clicking a non-editable block in the editor (and then outside)

### ISSUE THREAD

**Issue by Zhouzi** [ianstormtaylor/slate#1769]
Clicking an element with `contentEditable={false}` in the editor is ignored and the selection remains as if nothing happened. The selection says that the focused node is still the one that was focused before clicking the non-editable element. More importantly, clicking outside of the editor after that doesn't clear the selection which still says that it's focused.

## Steps to reproduce

1. Create a document with a node that renders a `contentEditable={false}` component
2. Click in the editor so `value.selection.isFocused` becomes `true`
3. Click on the `contentEditable={false}`, `value.selection.isFocused` remains true
4. Click somewhere outside of the editor, `value.selection.isFocused` is still true

[Reproduction](https://codesandbox.io/s/6j2v26y3xn)

*The bug was not present in an older version of Slate: [reproduction](https://codesandbox.io/s/24kzl9w5kr).*

**Comment by SamyPesse** [comment]
@ianstormtaylor Any idea where this regression may come from ?

@Zhouzi Which versions are being used in both reproductions ? (it may help us find the regression)

**Comment by ianstormtaylor** [comment]
Nope, not off the top of my head. Would need someone to investigate.

**Comment by nucleartux** [comment]
@ianstormtaylor  Any progress? I have same problem with input inside void block (as in example https://github.com/ianstormtaylor/slate/blob/master/examples/embeds/video.js), when I click on input `value.blocks` still contains previous focused element.
